test.ts

```diff
@@ -5,22 +5,47 @@ import {PluginItem, transform, TransformOptions} from '@babel/core'
 
 import preset from '.'
 
-const ENVS = ['development', 'production', 'esm', 'cjs']
+const NON_TEST_ENVIRONMENTS = [
+  'development',
+  'production',
+  'esm',
+  'cjs',
+  'development-modern',
+  'production-modern',
+]
 
 expect.addSnapshotSerializer({
   test: (val: unknown) => typeof val === 'string',
   print: (val: string) => val,
 })
 
-function macro(title: string, fixture: string, presetOptions = {}): void {
+function macro(
+  title: string,
+  fixture: string,
+  presetOptions = {},
+  macroOptions = {
+    environments: NON_TEST_ENVIRONMENTS,
+  }
+): void {
   test(`${title} given ${JSON.stringify(presetOptions)}`, () => {
     const file = `${__dirname}/fixtures/${fixture}`
     const input = fs.readFileSync(file, 'utf8')
     expect(input).toMatchSnapshot('input')
 
-    ENVS.forEach((envName) => {
+    const {environments} = macroOptions
+    environments.forEach((envName) => {
       const presets: PluginItem[] = [[preset, presetOptions]]
-      const options: TransformOptions = {envName, presets, filename: `/${fixture}`, babelrc: false}
+      const options: TransformOptions = {
+        envName,
+        presets,
+        filename: `/${fixture}`,
+        babelrc: false,
+        // TODO: check if browserslist can be moved to the fixtures/ directory once https://github.com/babel/babel/pull/13031 is merged
+        // TODO: remove @ts-ignore once typedefs catch up with API
+        // Note: this option seems to be necessary even for config files at project root
+        // @ts-ignore
+        browserslistConfigFile: `${__dirname}/.browserslistrc`,
```

Comment:
I looked into this and it turns out to not be working as expected.

This is being passed as a top-level option to the Babel compiler whereas our preset is passing `browserslistEnv` to `@babel/preset-env`, _not_ the compiler. `@babel/preset-env` uses the options that were passed to the compiler if it hasn't received its own options, but when it has, it completely ignores those of the compiler.

The result is that this line has no effect when `env.includes('modern')`. It's just using the default resolution logic, which is to ascend the directory tree, which is why this works when the file is here but not when it's placed in the fixtures directory. (Note that when it's placed in the fixtures directory, it does still work for non-modern envs, because, in that case, our preset is passing `browserslistEnv: undefined` to `@babel/preset-env`. You can verify this by moving `.browserslistrc` into the fixtures directory and updating the `production` env to target only `last 1 Chrome version`. The snapshots show that `production` gets a modern transpilation whereas `production-modern` transpiles the kitchen sink because no browserslist file was found.)

I'm going to push a fix that uses `@babel/preset-env`'s `configPath` option instead.

Reply:
Resolved in https://github.com/kensho-technologies/babel-preset/pull/88/commits/743a8a0d8c164d24a97e70913deb35b3ab3b429f.

Reply:
Nice!

One thought -- is spreading `...rest` into `preset-env`'s config what we want to ship in the end? If by chance a param gets added to our preset's API that has meaning for `preset-env` (and we don't notice), or `preset-env` gets updated to understand one of our params, it could cause unintended side-effects.

Reply:
Actually ignore all those arguments. I didn't think it through, sorry.